internal/rukpak/source/containers_image.go

```diff
@@ -45,6 +48,22 @@ func (i *ContainersImageRegistry) Unpack(ctx context.Context, bundle *BundleSour
 	if err != nil {
 		return nil, err
 	}
+
+	//////////////////////////////////////////////////////
+	//
+	// Verify that registries configuration is up-to-date
+	//
+	//////////////////////////////////////////////////////
+	data, err := os.ReadFile(srcCtx.SystemRegistriesConfPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read registries configuration file, error: %w", err)
+	}
+	registriesConfHash := sha256.Sum256(data)
+	if registriesConfHash != i.registriesConfHash {
+		i.registriesConfHash = registriesConfHash
```

Comment:
Pretty sure the `ContainersImageRegistry` struct is thread-safe and meant to be usable concurrently. With that in mind, we would need the entire "compute hash + update hash field" logic to be guarded by a mutex.

Reply:
Absolutely, that was a miss - I'll add that!